autopilot/src/autopilot.cpp

```diff
@@ -36,549 +36,566 @@
+{
+  if (last_heartbeat_stamp_ < ev.last_real)  // no heartbeat received in the past period
   {
-    setRPM.commanded_rpms = 0.0;
-    thrusterPub.publish(setRPM);
+    active_setpoints_ = Setpoint::Pitch;
+    desired_pitch_ = -max_ctrl_fin_angle_;
+    ROS_INFO_THROTTLE(5.0, "Mission control is down");
   }
 }
 
-void AutoPilotNode::stateCallback(const auv_interfaces::StateStamped& msg)
+void AutoPilotNode::missionHeartbeatCallback(const mission_control::ReportHeartbeat& msg)
 {
-  boost::mutex::scoped_lock lock(m_mutex);
-
-  currentDepth = msg.state.manoeuvring.pose.mean.position.z;
-  currentRoll = radiansToDegrees(msg.state.manoeuvring.pose.mean.orientation.x);
-  currentPitch = radiansToDegrees(msg.state.manoeuvring.pose.mean.orientation.y);
-  currentYaw = radiansToDegrees(msg.state.manoeuvring.pose.mean.orientation.z);
-  geodesy::fromMsg(msg.state.geolocation.position, currentPosition);
+  last_heartbeat_stamp_ = msg.header.stamp;
+  ROS_DEBUG_THROTTLE(5.0, "Mission control is up");
 }
 
-void AutoPilotNode::missionStatusCallback(const mission_control::ReportExecuteMissionState& data)
+void AutoPilotNode::missionStatusCallback(const mission_control::ReportExecuteMissionState& msg)
 {
-  using mission_control::ReportExecuteMissionState;
-  if (data.execute_mission_state == ReportExecuteMissionState::COMPLETE)
+  if (msg.execute_mission_state == mission_control::ReportExecuteMissionState::COMPLETE)
   {
-    boost::mutex::scoped_lock lock(m_mutex);
-    missionMode = true;
-
-    desiredRoll = 0;
-
-    // surface
-    desiredPitch = -maxCtrlFinAngle;
-    depthControl = false;
-    altitudeControl = false;
-
-    desiredRudder = 0;  // straight
-    fixedRudder = true;
-
-    desiredSpeed = 0;  // coast
+    active_setpoints_ = Setpoint::Pitch;
+    desired_pitch_ = -max_ctrl_fin_angle_;
   }
 }
 
-void AutoPilotNode::HandleActivateManualControl(const jaus_ros_bridge::ActivateManualControl& data)
+void AutoPilotNode::handleActivateManualControl(const jaus_ros_bridge::ActivateManualControl& msg)
 {
-  boost::mutex::scoped_lock lock(behaviorMutex);
-  if (data.activate_manual_control)
+  if (msg.activate_manual_control)
   {
-    // ROS_INFO("Jaus says its in control.");
-    autoPilotInControl = false;
-    missionMode = false;
-    // clear the following so we do not take off after manual control is released.
-    desiredSpeed = 0;
-    desiredRoll = 0;
-    desiredYaw = 0;
-    desiredPitch = 0;
-    desiredDepth = 0;
-    desiredRudder = 0;
-    fixedRudder = true;
-    depthControl = false;
-    altitudeControl = false;
+    if (auto_pilot_in_control_) {
+      ROS_INFO("Manual control activated!");
+      // clear the following so we do not take off after manual control is released.
+      active_setpoints_ = {};
+      auto_pilot_in_control_ = false;
+    }
   }
   else
   {
-    // ROS_INFO("Jaus says its not in control.");
-    autoPilotInControl = true;
+    if (!auto_pilot_in_control_) {
```

Comment:
nit: should be on a line by itself

Reply:
Indeed. See https://github.com/ChrisScianna/ROS-Underwater-RnD/pull/134/commits/b72ee3a4d2a1caa026f360e9708ca5cb5537bb7e.